src/cloudai/workloads/nemo_run/cloudai_nemorun.py

```
@@ -170,6 +173,62 @@ def timing_callback() -> run.Config[TimingCallback]:
+        if trainer.global_rank != 0:
+            return
+        try:
+            # Single failure domain: if any of the three roots raises (module without
+            # .config, non-Megatron strategy, datamodule None), the whole file is skipped.
+            cfg = {
+                "model": dataclasses.asdict(pl_module.config),
+                "parallelism": dataclasses.asdict(trainer.strategy.parallelism),
+                "data": vars(trainer.datamodule),
+            }
+            Path("/cloudai_run_results/nemo_config.json").write_text(json.dumps(cfg, indent=2, default=str))
+        except Exception as e:  # reporting must never break training
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)